Added local transcription support using whisper-timestamped library

hanifalisohag opened the pull request:
Added #7 local support for transcriptions

bryananderson commented:
I am fine adding local Whisper as an option at startup of the demo script, but I want to leave the API as an option too. The reason is that in my testing (on nvidia a10), local is about 5x slower. Can you make it configurable?

Other notes

- I am fine adding Docker/Podman instructions to the readme, but can you remove the other unrelated changes? As well as the changelog-style note about the local Whisper model. Probably we should just have a changelog section (TODO for myself).
- The new dependencies should be in the optional-dependencies section since they are only relevant to the demo script.

hanifalisohag commented:
> I am fine adding local Whisper as an option at startup of the demo script, but I want to leave the API as an option too. The reason is that in my testing (on nvidia a10), local is about 5x slower. Can you make it configurable?
> 
> Other notes
> 
> * I am fine adding Docker/Podman instructions to the readme, but can you remove the other unrelated changes? As well as the changelog-style note about the local Whisper model. Probably we should just have a changelog section (TODO for myself).
> * The new dependencies should be in the optional-dependencies section since they are only relevant to the demo script.
> * In the dockerfile, you should not need to separately `pip install -U huggingface-hub`, since this is already in pyproject.toml and thus will be installed as part of `pip install '.[demo]'`. Is there a reason you install it separately?

I have added both options: OpenAI API and Local Whisper (tiny model is the fastest with less accuracy, made it default instead of the turbo model is more accurate but slower) 

1. I have removed the other changes from the Readme. Only the docker commands are added for docker or podman build
2. I have moved the demo dependencies to the optional dependency section
3. Initially I was seeing issues with the huggingface-hub version mentioned on the pyproject.toml and after upgrading it solves the isseus. If that occurs again we might need to switch to the latest one.

bryananderson commented:
I rebased and cleaned up whitespace and merged [here](https://github.com/playht/PlayDiffusion/pull/14). Thanks @hanifalisohag !
